### PR TITLE
Param rework

### PR DESCRIPTION
**Pull request by LorenzMeier**
This puts the param subsystem into its own library.

### CODE REVIEW

File: src/lib/param/param.h [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+ *
+ * Parameters are represented by parameter handles, which can
+ * be obtained by looking up (or creating?) parameters.
+ */
+typedef uintptr_t	param_t;
+
+/**
+ * Handle returned when a parameter cannot be found.
+ */
+#define PARAM_INVALID	((uintptr_t)0xffffffff)
```
Comment (bansiesta) [inline src/lib/param/param.h]: Why this and not nullptr?

File: src/platforms/posix/drivers/airspeedsim/airspeedsim.cpp [hunk 2]
```diff
@@ -60,7 +60,7 @@
 
 #include <systemlib/airspeed.h>
 #include <systemlib/err.h>
-#include <systemlib/param/param.h>
+#include <lib/param/param.h>
```
Comment (bansiesta) [inline src/platforms/posix/drivers/airspeedsim/airspeedsim.cpp]: Why "lib/" ?

File: src/modules/commander/accelerometer_calibration.cpp [hunk 3]
```diff
@@ -143,7 +143,7 @@
 #include <drivers/drv_accel.h>
 #include <geo/geo.h>
 #include <conversion/rotation.h>
-#include <systemlib/param/param.h>
+#include <lib/param/param.h>
```
Comment (bansiesta) [inline src/modules/commander/accelerometer_calibration.cpp]: Again, why "lib/" ?

File: src/modules/systemlib/param/param.h [hunk 4]
```diff
@@ -1,417 +1,8 @@
...
- * cannot change type or size over its lifetime.  If any of these assumptions
- * are invalidated, the attributes should be re-evaluated.
- */
+#ifndef _PARAM_DEPRECATED_H
+#define _PARAM_DEPRECATED_H
 
-#ifndef _SYSTEMLIB_PARAM_PARAM_H
-#define _SYSTEMLIB_PARAM_PARAM_H
+#include <lib/param/param.h>
+#info This header is deprecated. Please include now <param/param.h> instead of the old #include <param/param.h>
```
Comment (bansiesta) [inline src/modules/systemlib/param/param.h]: Does this comment make sense?

File: src/lib/param/param.h (hunk 1 above)
Comment (bansiesta) [inline src/lib/param/param.h]: Ok, I see, this is not part of the PR. That's just how it has always been, fair enough.

**Comment by LorenzMeier** [comment]
Need to fix this up.

**Comment by LorenzMeier** [comment]
Will re-do this in time.
